docs/src/services/table-of-contents.tsx

```diff
@@ -35,17 +35,19 @@ export function useTableOfContents() {
 
 export function useTableOfContentsRegister({ title, level }: { title: string; level: number }) {
   const [, setTableOfContents] = React.useContext(TableOfContentsContext);
-  const id = React.useMemo(() => uuidv4(), []);
+  const idRef = React.useRef<string>("");
+  const id = uuidv4();
 
   React.useEffect(() => {
+    idRef.current = id;
```

Comment:
This current code is causing infinite loop, because on every re-render there is a new id. And if there's a new id, the useEffect is executed. And if the useEffect is executed, there's a re-render. And if there's a re-render, there's a new id. And it goes on a loop ➿ 

IMO, this line should be outside the useEffect, and the dependency should change from `id` to `idRef`. This way the useEffect is only executed when the ref changes (which is exactly what we don't expect to happen).

Reply:
Yeah, did not pay much attention to this one